examples/inspection/plot_partial_dependence.py

```diff
@@ -213,3 +225,61 @@
 plt.subplots_adjust(top=0.9)
 
 plt.show()
+
+##############################################################################
+# Individual Conditional Expectation (ICE) Plots
+# ----------------------------------------------
+#
+# One of the main limitations of ICE plots is that the plot gets overcrowded
+# if many ICE curves are drawn. Due to this, we will use a random sample of
+# 50 instances for the ICE plot.
+#
+# ICE plots are achieved by setting ``individual=True``.
+# Let's now compute the ICE plots for this Gradient Boosting model:
+
+X_train_sample = X_train.sample(50, random_state=0)
```

Comment:
It is where I think it is good to have a `subsample` parameter. You need to subsample but you still want to see the PD being the average over all the training sample and not only the subset. It something that currently a user cannot do easily.

Reply:
Added in https://github.com/scikit-learn/scikit-learn/pull/16619/commits/f6f2f3cf73d029f3b48c1f8617dcd3dba1d95f9c